### PR TITLE
Appointment edit

### PR DESCRIPTION
**Pull request by sowmika148**

### CODE REVIEW

File: ui/react-components/api/recurringAppointmentsApi.js [hunk 1]
```diff
@@ -8,3 +8,12 @@ export const saveRecurringAppointments = async data => {
         console.error(error);
     }
 };
+
+export const getRecurringAppointmentByUuid = async (appointmentUuid) => {
```
Comment (bsneha90) [inline ui/react-components/api/recurringAppointmentsApi.js]: can we rename this to reucurringAppointmentFor??

File: ui/react-components/api/recurringAppointmentsApi.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import axios from "axios";
-import {recurringAppointmentsSaveUrl} from "../constants";
+import {recurringAppointmentByUuidUrl, recurringAppointmentsSaveUrl} from "../constants";
```
Comment (bsneha90) [inline ui/react-components/api/recurringAppointmentsApi.js]: can we get recurringAppointmetn without UUID? why do we call the constant as ByUuidUrl?

File: ui/react-components/components/AddAppointment/AddAppointment.jsx [hunk 3]
```diff
@@ -35,21 +29,20 @@ import AppointmentDatePicker from "../DatePicker/DatePicker.jsx";
...
-    getYesterday,
-    isSpecialitiesEnabled,
-    maxAppointmentProvidersAllowed
+    getYesterday
 } from "../../helper.js";
 import {getSelectedWeekDays, getWeekDays} from "../../services/WeekDaysService/WeekDaysService";
 import ButtonGroup from "../ButtonGroup/ButtonGroup.jsx";
 import {getErrorTranslations} from "../../utils/ErrorTranslationsUtil";
 import {isEmpty} from 'lodash';
+import SearchFieldsContainer from "../SearchFieldsContainer/SearchFieldsContainer.jsx";
```
Comment (bsneha90) [inline ui/react-components/components/AddAppointment/AddAppointment.jsx]: Can we think of a better name?

File: ui/react-components/components/AddAppointment/AddAppointment.jsx [hunk 4]
```diff
@@ -256,51 +249,12 @@ const AppointmentEditor = props => {
...
-                        <LocationSearch onChange={(optionSelected) => updateAppointmentDetails({location: optionSelected.value})}/>
-                        <ErrorMessage message={undefined}/>
-                    </div>
-                </div>
-                <div className={classNames(searchFieldsContainerRight)} data-testid="provider-search">
-                    <ProviderSearch onChange={selectedProviders => updateAppointmentDetails({providers: selectedProviders})}
-                                    maxAppointmentProvidersAllowed={maxAppointmentProvidersAllowed(appConfig)}/>
-                </div>
-            </div>
+            <SearchFieldsContainer updateAppointmentDetails={updateAppointmentDetails} updateErrorIndicators={updateErrorIndicators}
```
Comment (bsneha90) [inline ui/react-components/components/AddAppointment/AddAppointment.jsx]: This makes it more difficult to understand the code?? Do we really need this?

File: ui/react-components/components/EditAppointment/EditAppointment.jsx [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+    const generateAppointmentDetails = async () => {
+        const appointment = isRecurring ? await getRecurringAppointmentByUuid(appointmentUuid) : await getAppointmentByUuid(appointmentUuid);
+        const appointmentResponse = isRecurring
+            ? (appointment && appointment.data && appointment.data.appointmentDefaultResponse) || undefined
+            : (appointment && appointment.data && appointment.data) || undefined;
+        const recurringPattern = isRecurring
+            ? (appointment && appointment.data && appointment.data.recurringPattern) || undefined : undefined;
+        if (appointmentResponse) {
+            updateAppointmentDetails({
```
Comment (bsneha90) [inline ui/react-components/components/EditAppointment/EditAppointment.jsx]: Can we move all the appointment specific things like these to an Appointment Model?? Talk to Alekhya and Siva also about this before doing anything.

File: ui/reactAngularAdaptor.js [hunk 6]
```diff
@@ -1,22 +1,25 @@
...
+function reactAddAppointmentController($scope, $state, $stateParams) {
     $scope.onBack = function () {
         $state.go('^', $state.params, {reload: true});
     };
 
     $scope.setViewDate = function (date) {
-      $state.params.viewDate = date;
+        $state.params.viewDate = date;
     };
+    console.log($state.current.url);
```
Comment (bsneha90) [inline ui/reactAngularAdaptor.js]: can we remove this console log?

Comment (bsneha90) [reply ui/reactAngularAdaptor.js]: You will get conflicts with Popup card Alekhya and Siva are working on, please coordinate.

Comment (sowmika148) [reply ui/reactAngularAdaptor.js]: Yup. removed

File: ui/react-components/api/recurringAppointmentsApi.js (hunk 1 above)
Comment (sowmika148) [reply ui/react-components/api/recurringAppointmentsApi.js]: Done

File: ui/react-components/api/recurringAppointmentsApi.js (hunk 2 above)
Comment (sowmika148) [reply ui/react-components/api/recurringAppointmentsApi.js]: Done

File: ui/react-components/components/EditAppointment/EditAppointment.jsx (hunk 5 above)
Comment (sowmika148) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: Will be handled later together for add and edit

File: ui/react-components/components/AddAppointment/AddAppointment.jsx (hunk 4 above)
Comment (sowmika148) [reply ui/react-components/components/AddAppointment/AddAppointment.jsx]: This snippet is common Add and Edit appointment. So we extracted that part.

**Comment by bsneha90** [comment]
New PR raised with conflicts resolved, so closing this PR.

**Pull request closed without merging**